python/langsmith/_internal/_background_thread.py

```diff
@@ -67,16 +94,20 @@ def _tracing_thread_handle_batch(
     batch: List[TracingQueueItem],
     use_multipart: bool,
 ) -> None:
-    create = [it.item for it in batch if it.action == "create"]
-    update = [it.item for it in batch if it.action == "update"]
-    feedback = [it.item for it in batch if it.action == "feedback"]
     try:
+        ops = combine_serialized_queue_operations([item.item for item in batch])
         if use_multipart:
-            client.multipart_ingest(
-                create=create, update=update, feedback=feedback, pre_sampled=True
-            )
+            client._multipart_ingest_ops(ops)
         else:
-            client.batch_ingest_runs(create=create, update=update, pre_sampled=True)
+            if any(isinstance(op, SerializedFeedbackOperation) for op in ops):
+                logger.warn(
+                    "Feedback operations are not supported in non-multipart mode"
+                )
+                ops = [
```

Comment:
silly but you could do this first and then compare lenghts if you only wanted to go over once